app/controllers/report_controller/reports.rb

```diff
@@ -196,13 +197,17 @@ def menu_repname_update(old_name, new_name)
         lvl1[1].each_with_index do |lvl2, j|
           lvl2[1].each_with_index do |rep, k|
             if rep == old_name
-              menu[i][1][j][1][k] = new_name
+              if new_name.nil?
+                menu[i][1][j][1].delete_at(k)
```

Comment:
deleting from an array that you are iterating skips elements.

But in this case it will only ever have 1 entry, so it is probably fine